uniffi_macros/src/fnsig.rs

```diff
@@ -374,6 +406,20 @@ impl NamedArg {
     }
 }
 
+fn looks_like_result(return_type: &ReturnType) -> bool {
+    if let ReturnType::Type(_, ty) = return_type {
+        if let Type::Path(p) = &**ty {
+            if let Some(seg) = p.path.segments.last() {
+                if seg.ident == "Result" {
```

Comment:
You mentioned in a different comment that this would fail for aliased results, but am I reading wrong that this would still work for the most comment aliased result of `Result<T>`?

Reply:
I think it would, but it doesn't matter either way.  This is only used to implement the UDL-only behavior of automatically mapping errors using Into (#1749).  So we only need to make sure that the UDL-based generator outputs the expected tokens.

Depending on the decision in 1749, we could consider using this for proc-macros as well, but I think there's probably a way that works without parsing tokens.